### ISSUE TITLE
[Analytics] Events around sign up/account creation

### ISSUE THREAD

**Issue by anipetrov** [artsy/eigen#1123]
I used the [Segment debugger](https://segment.com/artsy-engineering/eigen-dev/debugger0) tab for this troubleshooting but it is not exactly real-time + it is challenging if other people are on the beta at the moment. Let me know if you think there are better ways to go about figuring out which event fires when exactly. Here are some initial observations

1. `Sign up with email` event fires even if there was an error signing up; if consecutively you do sign up successfully, the sign up event is fired again
2. After a successful sign up, the  `Sign up with email` event is fired more than once
3. `Created account` event does not fire if onboarding is skipped

cc @katarinabatina

**Comment by alloy** [comment]
@anipetrov Want to dip your toes in some play time with Xcode and the debugger (specifically breakpoints) to be able to track these down yourself?

**Comment by anipetrov** [comment]
That would be amazing @alloy, thank you so much for offering!!

On Monday, February 8, 2016, Eloy Durán <notifications@github.com> wrote:

> @anipetrov <https://github.com/anipetrov> Want to dip your toes in some
> play time with Xcode and the debugger (specifically breakpoints) to be able
> to track these down yourself?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artsy/eigen/issues/1123#issuecomment-181310907>.
>


-- 
[A.]      Ani Petrov

           Artsy <https://artsy.net/>
           ani@artsy.net
           +1 609 216 4575

**Comment by alloy** [comment]
We’re gonna take it one step further and write some code: https://github.com/artsy/eigen/issues/1127

Can you make a cal event for when you’d have half a day in your morning (my afternoon) to spend on this? If you can wait till the first week of March, then we can even do it at the office when I’m over.

**Comment by anipetrov** [comment]
I could do either -- whatever you think would be the best use of your time. If you'd rather not waist half a day of your time here on this, I will put a cal event for some time next week. 

Thank you again for offering to take the time!!

**Comment by alloy** [comment]
Au contraire, I imagine that pairing whilst I’m there would be more efficient! (And let’s be honest, otherwise m time is filled with coffee drinking anyways.) So let’s do it then :+1:

**Comment by anipetrov** [comment]
Amazing, thanks a million! I will put something on your cal!

**Comment by alloy** [comment]
:bow: :+1:

**Comment by orta** [comment]
Gonna take this I'm afraid, you two should pair on the analytics visual mode :P

- [ ] Sign up with email event fires even if there was an error signing up; if consecutively you do sign up successfully, the sign up event is fired again
- [ ] After a successful sign up, the Sign up with email event is fired more than once
- [ ] Created account event does not fire if onboarding is skipped

**Comment by orta** [comment]
I've done the top/bottom pretty easily - the sign up event stuff is complicated by the fact that we use 2 types of events to cover 3 types of cases. We need to cover tapping the buttons, actually signing in and creating a new user. I think some of the analytics around these are mixed up, will double check it all over.

**Comment by alloy** [comment]
@orta Is that TODO list still up-to-date and the double-fire still needs to be fixed?

**Comment by orta** [comment]
I think so, yes

**Comment by alloy** [comment]
Ok :+1:

**Comment by alloy** [comment]
The [`Sign up with email`](https://github.com/artsy/eigen/blob/950639a9f15410a8b8043c4f7cb950dc18560231/Artsy/App/ARAnalyticsConstants.m#L24) event is currently being fired from 3 different locations:

* `createUserWithName:email:` https://github.com/artsy/eigen/blob/84a274ade261b4e32b97881e0299f4a3ce8ff950/Artsy/Networking/API_Modules/ARUserManager.m#L387
* `createUserViaFacebookWithToken:email:name:` https://github.com/artsy/eigen/blob/84a274ade261b4e32b97881e0299f4a3ce8ff950/Artsy/Networking/API_Modules/ARUserManager.m#L446
* `createUserViaTwitterWithToken:secret:email:name:` https://github.com/artsy/eigen/blob/84a274ade261b4e32b97881e0299f4a3ce8ff950/Artsy/Networking/API_Modules/ARUserManager.m#L480

However, in the first case there’s an additional [`Created account`](https://github.com/artsy/eigen/blob/84a274ade261b4e32b97881e0299f4a3ce8ff950/Artsy/Networking/API_Modules/ARUserManager.m#L406) event but no initial ‘start’ event, whereas in the other two cases there are _only_ initial ‘sign up with facebook/twitter’ events and no ‘created’ event.

In short, I think we need:

* ‘Sign-up with email’ -> ‘created account’
* ‘Connect with twitter’ -> ‘created account’
* ‘Connect with facebook’ -> ‘created account’

Does that make sense?

**Comment by alloy** [comment]
With regards to the double firing of the sign-up event when successfully creating an account, I haven’t yet figured out where/how that is triggered.

**Comment by alloy** [comment]
@anipetrov In case the above makes sense, I have this PR #1301.

**Comment by alloy** [comment]
I had not expected the link from my PR would auto-close this.

**Comment by alloy** [comment]
@orta if you have info on where the double firing of the event is/was coming from, please let me know.

**Comment by orta** [comment]
I can't re-procude the double firings, it's very possible I fixed it but did't tick the box

**Comment by alloy** [comment]
Ok 👍
